### PR TITLE
Add jsdocs.

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Documents a few of the sign and verify related functions.

### CODE REVIEW

File: lib/ProofSet.js [hunk 1]
```diff
@@ -150,39 +150,34 @@ module.exports = class ProofSet {
...
    *   `documentLoader`) or a plain object (JSON-LD document).
-   * @param options {object} Options hashmap.
-   *
-   * A `suite` option is required:
-   *
-   * @param options.suite {LinkedDataSignature or Array of LinkedDataSignature}
-   *   acceptable signature suite instances for verifying the proof(s).
    *
-   * A `purpose` option is required:
+   * @param {LinkedDataSignature|Array<LinkedDataSignature>} suite -
```
Comment (mattcollier) [inline lib/ProofSet.js]:
```suggestion
   * @param {LinkedDataSignature|LinkedDataSignature[]} suite -
```
Thanks for cleaning up the docs!

This exact usage is described under `Allows one type OR another type (type union)`
https://jsdoc.app/tags-param.html

File: lib/jsonld-signatures.js [hunk 2]
```diff
@@ -14,6 +14,33 @@ Object.assign(api, constants);
 // TODO: support `ProofChain`
 const ProofSet = require('./ProofSet');
 
+/**
+ * @param {object|string} document - Object to be signed, either a string URL
+ *  (resolved to an object via the given `documentLoader`) or a plain object
+ *  (JSON-LD document).
+ *
+ * @param {ProofPurpose} purpose
```
Comment (mattcollier) [inline lib/jsonld-signatures.js]: needs a description.

File: lib/jsonld-signatures.js [hunk 3]
```diff
@@ -14,6 +14,33 @@ Object.assign(api, constants);
...
+ *
+ * Advanced optional parameters and overrides:
+ *
+ * @param {function} [documentLoader]  a custom document loader,
```
Comment (mattcollier) [inline lib/jsonld-signatures.js]:
```suggestion
 * @param {function} [documentLoader]  - A custom document loader,
```

File: lib/jsonld-signatures.js [hunk 4]
```diff
@@ -14,6 +14,33 @@ Object.assign(api, constants);
 // TODO: support `ProofChain`
 const ProofSet = require('./ProofSet');
 
+/**
+ * @param {object|string} document - Object to be signed, either a string URL
```
Comment (mattcollier) [inline lib/jsonld-signatures.js]:
```suggestion
 * @param {object|string} document - The document to be signed, either a string URL
```

File: lib/jsonld-signatures.js [hunk 5]
```diff
@@ -14,6 +14,33 @@ Object.assign(api, constants);
 // TODO: support `ProofChain`
 const ProofSet = require('./ProofSet');
 
+/**
```
Comment (mattcollier) [inline lib/jsonld-signatures.js]: Typically the jsdoc has a one liner for a function description.

File: lib/jsonld-signatures.js [hunk 6]
```diff
@@ -32,6 +59,42 @@ api.sign = async function sign(document, {
   }
 };
 
+/**
```
Comment (mattcollier) [inline lib/jsonld-signatures.js]: function description.

File: lib/jsonld-signatures.js [hunk 7]
```diff
@@ -32,6 +59,42 @@ api.sign = async function sign(document, {
...
+ * @param {object|string} document - Object with one or more proofs to be
```
Comment (mattcollier) [inline lib/jsonld-signatures.js]:
```suggestion
 * @param {object|string} document - The document with one or more proofs to be
```

File: lib/jsonld-signatures.js [hunk 8]
```diff
@@ -32,6 +59,42 @@ api.sign = async function sign(document, {
...
+ *   verified, either a string URL (resolved to an object via the given
+ *   `documentLoader`) or a plain object (JSON-LD document).
+ *
+ * @param {LinkedDataSignature|Array<LinkedDataSignature>} suite -
```
Comment (mattcollier) [inline lib/jsonld-signatures.js]:
```suggestion
 * @param {LinkedDataSignature|LinkedDataSignature[]} suite -
```

File: lib/jsonld-signatures.js [hunk 9]
```diff
@@ -32,6 +59,42 @@ api.sign = async function sign(document, {
...
+ *   Acceptable signature suite instances for verifying the proof(s).
+ *
+ * @param {ProofPurpose} purpose - A proof purpose instance that will
+ *   match proofs to be verified and ensure they were created according to
+ *   the appropriate purpose.
+ *
+ * Advanced optional parameters and overrides:
+ *
+ * @param {function} [documentLoader]  a custom document loader,
```
Comment (mattcollier) [inline lib/jsonld-signatures.js]:
```suggestion
 * @param {function} [documentLoader]  - A custom document loader,
```

File: lib/purposes/ControllerProofPurpose.js [hunk 10]
```diff
@@ -32,6 +32,22 @@ module.exports = class ControllerProofPurpose extends ProofPurpose {
     }
   }
 
+  /**
+   * Validates the purpose of a proof. This method is called during
+   * proof verification, after the proof value has been checked against the
+   * given verification method (e.g. in the case of a digital signature, the
+   * signature has been cryptographically verified against the public key).
+   *
+   * @param proof
```
Comment (mattcollier) [inline lib/purposes/ControllerProofPurpose.js]: seems like maybe a @typedef or something is in order so there is no need to repeat what a `documentLoader` or `expansionMap` are 3 or 4 times.

File: lib/ProofSet.js (hunk 1 above)
Comment (dmitrizagidulin) [reply lib/ProofSet.js]: Awesome, I like that notation even better!

File: lib/purposes/ControllerProofPurpose.js (hunk 10 above)
Comment (dmitrizagidulin) [reply lib/purposes/ControllerProofPurpose.js]: Ah yeah, I need to look into those..

File: lib/jsonld-signatures.js (hunk 6 above)
Comment (dmitrizagidulin) [reply lib/jsonld-signatures.js]: Added, thanks!

File: lib/jsonld-signatures.js (hunk 5 above)
Comment (dmitrizagidulin) [reply lib/jsonld-signatures.js]: Added.

File: lib/jsonld-signatures.js (hunk 6 above)
Comment (dmitrizagidulin) [reply lib/jsonld-signatures.js]: Added, thanks.

**Comment by dmitrizagidulin** [comment]
Thanks @mattcollier! I applied your changes (and squashed the commits a bit). The only things that remain are to use `@typedef`s, which I need to go learn about.

**Comment by aljones15** [comment]
@dmitrizagidulin do the new jsdoc comments get past the jsdoc linter? https://github.com/digitalbazaar/jsonld-signatures/pull/64

maybe we can merge the linter in this branch first?

**Comment by davidlehn** [comment]
The jsdoc linting rules may need to be tweaked so as not to be too strict.  Some docs now would be better than holding off to fix all the linting issues.

File: lib/jsonld-signatures.js [hunk 11]
```diff
@@ -32,6 +66,44 @@ api.sign = async function sign(document, {
...
+ * @param {boolean} [compactProof=true] - Indicates that this method cannot
+ *   assume that the incoming document has defined all proof terms in the
+ *   same way as the `constants.SECURITY_CONTEXT_URL` JSON-LD `@context`.
+ *   This means that this method must compact any found proofs to this
+ *   context for internal and extension processing; this is the default
+ *   behavior. To override this behavior and optimize away this step because
+ *   the caller knows that the input document's JSON-LD `@context` defines
+ *   the proof terms in the same way, set this flag to `false`.
+ *
+ * @return {Promise<{verified: boolean, results: Array, error: *}>} resolves
```
Comment (mattcollier) [inline lib/jsonld-signatures.js]: @dmitrizagidulin this needs to be updated to latest API.

Comment (dmitrizagidulin) [reply lib/jsonld-signatures.js:101]: @mattcollier What's the latest API? (I just double checked, and the signature seems to be the same?)

**Comment by dmitrizagidulin** [comment]
(Rebased on latest `master`.)

File: lib/purposes/ControllerProofPurpose.js (hunk 10 above)
Comment (dmitrizagidulin) [reply lib/purposes/ControllerProofPurpose.js:41]: @mattcollier I'd like to deal with `@typedef` in another PR, if at all possible.

File: lib/jsonld-signatures.js (hunk 11 above)
Comment (mattcollier) [reply lib/jsonld-signatures.js:102]: See: https://github.com/digitalbazaar/jsonld-signatures/blob/master/CHANGELOG.md#changed-1

Comment (dmitrizagidulin) [reply lib/jsonld-signatures.js:102]: @mattcollier Ahhh thank you!

Comment (dmitrizagidulin) [reply lib/jsonld-signatures.js]: Should be updated now.

**Comment by dmitrizagidulin** [comment]
(Since this is a comments-only PR, ignoring the Node 8 ci error.) Merging.

**Pull request merged**